packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1431,32 +1431,32 @@ export class MergeTree {
         if (!segoff.segment || !isRemovedAndAcked(segoff.segment)) {
             return segoff;
         }
-        // Slide to the next farthest valid segment in the tree. If no such segment is found
-        // slide to the last valid segment.
-        // TODO this walks the whole tree to find the segment - could write a more efficient
-        // walk that starts at the segment
-        let foundStart = false;
-        let foundSegmentPastStart = false;
         let slideToSegment: ISegment | undefined;
-        this.walkAllSegments(this.root, (seg) => {
+        const goFurtherToFindSlideToSegment = (seg) => {
             if (seg.seq !== UnassignedSequenceNumber && !isRemovedAndAcked(seg)) {
                 slideToSegment = seg;
-                if (foundStart) {
-                    foundSegmentPastStart = true;
-                    return false;
-                }
-            }
-            if (!foundStart && seg === segoff.segment) {
-                foundStart = true;
+                return false;
             }
             return true;
-        });
-        let offset = 0;
-        if (slideToSegment && !foundSegmentPastStart) {
+        };
+        // Slide to the next farthest valid segment in the tree.
+        this.rightExcursion(segoff.segment, goFurtherToFindSlideToSegment);
+        if (slideToSegment) {
+            return { segment: slideToSegment, offset: 0 };
+        }
+        // If no such segment is found, slide to the last valid segment.
+        this.leftExcursion(segoff.segment, goFurtherToFindSlideToSegment);
+
+        // Workaround TypeScript issue (https://github.com/microsoft/TypeScript/issues/9998)
+        const identity = (seg: ISegment | undefined) => seg;
```

Comment:
nit: I'd prefer casting; no need to add runtime behavior for ts deficiency. Not like this is a very hot code path though, so this is more the principle of the matter

Reply:
Fixed.